### DISCUSSION TITLE
extended syslog_sink with support for all syslog log levels

### DISCUSSION THREAD

**Discussion by rsniezek** [gabime/spdlog#2765] [Ideas]
Hi all,

How would you feel about adding a extended syslog sink that would allow user to log to syslog with all syslog log levels. The syslog sink that is provided right now is lacking: LOG_NOTICE, LOG_ALERT and LOG_EMERG and some users might want to use those as well.

How that could be done:

1. based on the syslog_sink
3. sink defines missing syslog log_levels as static constexpr level::level_enum variables that will be passed by client to logger->log(level) argument.
4. client wanting to use the extended levels uses the variables defined in sink when calling log on a logger with the extended sink
5. the mapping of the newly defined log levels to syslog severities is done in the sink class, similarily to how the syslog_sink does it, just with all values mapped correctly. This can be done either with larger std::array or with std::unordered_map
6. we don't define any new notice(), emergency(), alert() calls, extened severities are usable only via the log() call
I have working example that I will share once I have greenlight from my organization.

This way, we would have full syslog severities with just extra sink, no impact on any common part of the library. The usage would of course be restricted, e.g. global set_level would have to be used with caution to not try to pass new log levels to non-supporting sinks. We would also have limited filtering, because the new log levels would be higher prio than the critical.

Anyway, please let me know how do you feel about it or if you have better ideas how to approach this, it would be welcomed as well.

Cheers!

**Comment by suri-log** [comment]
yes this is missing piece from  SPD when people are trying to map between syslog and SPD log levels
